experimenter/experimenter/cirrus/tests/test_middleware.py

```diff
@@ -0,0 +1,109 @@
+from unittest import mock
+
+import pytest
+from django.test import TestCase
+from django.http import QueryDict
+
+from experimenter.cirrus.middleware import CirrusMiddleware
+
+
+class CirrusMiddlewareTests(TestCase):
+    def setUp(self):
+        self.response = "Response"
+
+        self.request = mock.Mock()
+        self.request.path = "/some/path"
+        self.request.user.id = "test"
+        self.request.GET = QueryDict()
+
+        self.cirrus_response = {"Features": {}, "Enrollments": []}
+
+    def test_without_user(self):
+        del self.request.user
+
+        with self.settings(CIRRUS_URL="cirrus"):
+            middleware = CirrusMiddleware(lambda request: self.response)
+
+        with mock.patch(
+            "experimenter.cirrus.middleware.requests.get"
+        ) as mock_requests_get:
+            response = middleware(self.request)
```

Comment:
Any reason not to use RequestFactory and the associated Django machinery?

Reply:
self.client.get(url, **kwargs) might add middleware to the request but im not sure